src/sync/probspecs.nim

```diff
@@ -56,27 +57,33 @@ proc exercisesWithCanonicalData(repo: ProbSpecsRepo): seq[ProbSpecsRepoExercise]
 func slug(repoExercise: ProbSpecsRepoExercise): string =
   extractFilename(repoExercise.dir)
 
-func initProbSpecsTestCase(node: JsonNode): ProbSpecsTestCase =
-  result.json = node
+func initProbSpecsTestCase(node: JsonNode, parents: seq[JsonNode]): ProbSpecsTestCase =
+  ProbSpecsTestCase(json: node, parents: parents)
 
 proc uuid*(testCase: ProbSpecsTestCase): string =
   testCase.json["uuid"].getStr()
 
 proc description*(testCase: ProbSpecsTestCase): string =
-  testCase.json["description"].getStr()
+  for parent in testCase.parents:
+    if parent.contains("description"):
```

Comment:
If there is no matching, there is only one parent and it does not contain a description. I could also rewrite this to do, which should also work:

```suggestion
  if testCase.parents.len > 2:
```

The current version is a bit more fool-proof, should any nesting ever forget to include the `description`.